plugins/node/opentelemetry-instrumentation-winston/src/instrumentation.ts

```diff
@@ -163,44 +186,96 @@ export class WinstonInstrumentation extends InstrumentationBase {
     return (original: Winston2LogMethod) => {
       const instrumentation = this;
       return function patchedLog(
-        this: unknown,
+        this: never,
         ...args: Parameters<typeof original>
       ) {
-        const span = trace.getSpan(context.active());
-
-        if (!span) {
-          return original.apply(this, args);
-        }
-
-        const spanContext = span.spanContext();
+        let record: Record<string, any> = {};
+        const config = instrumentation.getConfig();
+        if (!config.disableLogCorrelation) {
+          const span = trace.getSpan(context.active());
+          if (span) {
+            const spanContext = span.spanContext();
+            if (isSpanContextValid(spanContext)) {
```

Comment:
also here, previous to this PR, when `!isSpanContextValid(spanContext)` we would `return original.apply(this, args);` and skip calling the hook and injecting anything. now we will do it even for invalid contexts